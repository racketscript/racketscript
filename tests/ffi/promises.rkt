#lang racketscript/base
(require "promise-handlers.rkt")

;; see also async.rkt

($> (mkpromise)
    (then handler1)
    (catch errhandle)
    (then handler2))

;; ## prints:
;; Initial
;; handler1:
;;   do that (from catch)
;; handler2:
;;   do this no matter what (from last then)
