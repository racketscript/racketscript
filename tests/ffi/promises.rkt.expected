Initial
handler 1
  do that
handler 2
  do this no matter what
